sample-demo-lambda-app/kinesis_function.py

```diff
@@ -0,0 +1,110 @@
+import boto3
+import subprocess
+
+successes = 0
+
+# Dummy AWS Handler to kick off high level processes
+def lambda_handler(source_region, destination_region, credentials):
+
+    session = boto3.Session()
+
+    # Load Records into KINESIS
+    CLIENT_NAME = 'kinesis'
+    kinesis = session.client(CLIENT_NAME, region_name=source_region, aws_access_key_id=credentials,
+                             aws_secret_access_key=credentials['SecretAccessKey'],
+                             aws_session_token=credentials['SessionToken'])
+    process_kinesis(kinesis, "some_file_path.txt")
+
+    # Get SNS Topic ARNs
+    CLIENT_NAME = 'sns'
+    for region in [source_region, destination_region]:
+        sns = session.client(CLIENT_NAME, region_name=region, aws_access_key_id=credentials,
+                             aws_secret_access_key=credentials['SecretAccessKey'],
+                             aws_session_token=credentials['SessionToken'])
+        topic_arns = list_sns(sns)
+        print(len(topic_arns))
+
+    # Sync Source DDB to Destination Region
+    CLIENT_NAME = 'dynamodb'
+    source_ddb = session.client(CLIENT_NAME, region_name=source_region, aws_access_key_id=credentials['AccessKeyId'],
+                              aws_secret_access_key=credentials['SecretAccessKey'],
+                              aws_session_token=credentials['SessionToken'])
+
+    destination_ddb = session.client(CLIENT_NAME, region_name=destination_region)
+    sync_ddb_table(source_ddb, destination_ddb)
+
+
+# Scan returns paginated results, so only partial data will be copied
+def sync_ddb_table(source_ddb, destination_ddb):
+    response = source_ddb.scan(
+        TableName="table1"
+    )
+    for item in response['Items']:
+        destination_ddb.put_item(
+            TableName="table2",
+            Item=item
+        )
+
+# This code uses a mutable default argument and modifies it to return. This would leak results across calls
+def list_sns(sns, topics=[]):
+    response = sns.list_topics()
+    for topic_arn in response["Topics"]:
+        topics.append(topic_arn["TopicArn"])
+    return topics
+
+
+# Infinite loop because a list is modified while being iterated over, Indices are not updated.
+def infinite_loop():
+    words = ['aws', 'amazon', 'codeguru']
+    for w in words:
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It appears that you are modifying a sequence while iterating over it. The iterable object for the loop expression is calculated once and remains unchanged despite any index changes caused by the sequence modification. This might lead to unexpected bugs. If you need to modify the sequence, it is recommended that you first make a copy, such as by using slice notation.

[Learn more](https://docs.python.org/3/reference/compound_stmts.html#the-for-statement)